site-src/geps/gep-1282.md

```diff
@@ -98,12 +106,297 @@ Of course, from the section above, you can see that there are many other feature
+}
+
+// BackendTargetReference identifies an API object to apply BackendCapabilities to. 
+// 
+type BackendTargetReference struct {
+    // Group is the group of the target resource.
+    Group Group `json:"group"`
+
+    // Kind is kind of the target resource.
+    // Valid values include:
+    // * "Service" (the default if no Kind is specified)
+    // * "ServiceImport"
+    //
+    // Invalid values include:
+    //
+    // * "invalid/kind" - "/" is an invalid character
+    // +required
+    Kind string `json:"kind"`
+
+    // Name is the name of the target resource.
+    Name ObjectName `json:"name"`
+}
+
+// BackendCapabilitiesSpec defines a group of specific properties held by its
+// affiliated backend, and specified by the backend owner (typically the 
+// application developer).  
+//
+// Examples of properties that belong in the BackendCapabilitiesSpec include:
+// TLS re-encryption, session state, timeouts, retries, Websockets, HTTP/2, 
+// load balancer algorithm, healthcheck info, external traffic policy,
+// connection limits, error page, protocol switch, alerts, and others. 
+//
+// Support: Extended
+//
+type BackendCapabilitiesSpec struct {
+    // Backend refers to the object to which we apply these capabilities
+	// +required
+	Backend backendTargetReference `json:"backend,omitempty"`
+	
+    // TLS is the section of details on encryption processing in the traffic 
+    // hop from Gateway to backend, sometimes known as upstream TLS or re-encrypt.
+    // A xRoute configured as passthrough cannot specify a Forward encrypt type.
+    // TLS is useful for HTTPRoute and should not be used on UDP/TCP/TLSRoute.
+    //
+    // +optional
+    TLS *EncryptType `json:"tls,omitempty"`
+}
+
+// EncryptType describes the details needed to implement encryption from the
+// gateway to the backend.
+type EncryptType struct {
+    // The minimal TLSVersion that is accepted by the backend.
+    // If empty, there is no minimal backend TLSVersion.
+    // +optional	
+    MinVersion   *TLSVersion `json:"minVersion,omitempty"`
+
+    // The maximum TLSVersion that is accepted by the backend.
+    // If empty, there is no maximum backend TLSVersion.
+    // +optional    
+    MaxVersion   *TLSVersion  `json:"maxVersion,omitempty"`
+    
+    // CipherSuite is the minimum cipher suite that the backend accepts.
+    // If empty, anything that is appropriate for the TLS version is accepted.
+    // +optional
+    CipherSuite   string `json:"cipherSuite,omitempty"`
+    
+    // GatewayCert is the certificate used by the Gateway to prove its identity to
+    // the backend, if the backend so requires.
+    // +optional   
+    GatewayCert  *SecretObjectReference `json:"gatewayCert,omitempty"`
+    
+    // BackendCerts are the CA certificates used by the gateway to authenticate
+    // with the backend. This certificate allows the Gateway to verify and trust the 
+    // backend certificate.
+    // +optional	
```

Comment:
what happens if not set? We don't validate, or use system root certs?

Reply:
For the CACerts key, I think that if that field is set, it should be the entire trust chain used for certificate authentication in both directions, and if not set, system root certs should be used. This will let users use Let's Encrypt certs from cert-manager or similar for their service if that's what they want.